bridge/docs/bridging.md

```diff
@@ -34,3 +34,94 @@ Again, a withdrawfee of 1 TFT will be taken, so make sure you send a larger amou
 The amount withdrawn from TF Chain will be sent to your Stellar wallet.
 
 Example: ![swap_to_stellar](swap_to_stellar.png)
+
+## Deeper look at how the TFChain Bridge works
+
+### STELLAR -> TFCHAIN (Lock-and-Mint flow)
+In this section, we look into the details of transferring TFT from a Stellar Account to a TFChain Account. 
+
+1. A transaction is received on the bridge Stellar account and witnessed by a bridge validator..
+
+2. The transaction information undergoes validation. If it fails, a refund is issued (sent back to the stellar source account). now, We assume that the validation has passed, but we will examine the refund flow in the next section.
+
+3. A mint is proposed by calling `propose_or_vote_mint_transaction` extrinsic on the TFTBridgeModule in tfchain. This extrinsic inserts a new `MintTransaction` in `MintTransactions` storage that includes the `amount`, `target`, `block`, `votes`, and emits a `MintTransactionProposed` event. The mint is considered processed by the bridge side at this point.
+
+4. Other bridge validators also votes for that proposal by calling same extrinsic. If the `MintTransaction` exists in `MintTransactions` storage already, the extrinsic increments the `votes` count for the specified `MintTransaction` and emit `MintTransactionVoted` event.
+
+5. From the TFChain side, if the majority (more than the half) of bridge validators agree on the transaction, tokens are minted to the target address. This check happens every time the `propose_or_vote_mint_transaction` extrinsic is executed by validator call. Then, the transaction is removed from bridge pallet `MintTransactions` storage and added to `ExecutedMintTransactions`. Finally, a `MintCompleted` event is emitted.
+
+#### Overview of the TFChain Minting events
+1. `tftBridgeModule.MintTransactionProposed`: A bridge validator has proposed a mint transaction upon witness a stellar deposit.
```

Comment:
A bridge validator has proposed a mint transaction after being the first to report a stellar deposit.